### PR TITLE
[caldav] Copy recurrence properties during update

### PR DESCRIPTION
**Pull request by chriadam**
This commit ensures that recurrence information is correctly copied
into existing incidences during update.

### CODE REVIEW

**Comment by chriadam** [comment]
@blammit @pvuorela @pgerdt

File: src/caldavclient.cpp [hunk 1]
```diff
@@ -416,7 +416,7 @@ void CalDavClient::start()
         getSyncDateRange(mSyncStartTime, &fromDateTime, &toDateTime);
         notebooks = mStorage->notebooks();
     }
-    LOG_DEBUG("\n\n++++++++++++++ mSyncStartTime:" << mSyncStartTime << "LAST SYNC:" << lastSyncTime());
+    LOG_DEBUG("++++++++++++++ mSyncStartTime:" << mSyncStartTime << "LAST SYNC:" << lastSyncTime());
```
Comment (chriadam) [inline src/caldavclient.cpp]: unrelated, but journal doesn't handle newlines within debug output correctly

File: src/incidencehandler.cpp [hunk 2]
```diff
@@ -173,7 +173,7 @@ bool IncidenceHandler::todosEqual(const KCalCore::Todo::Ptr &a, const KCalCore::
     return true;
 }
 
-bool IncidenceHandler::journalsEqual(const KCalCore::Journal::Ptr &a, const KCalCore::Journal::Ptr &b)
+bool IncidenceHandler::journalsEqual(const KCalCore::Journal::Ptr &, const KCalCore::Journal::Ptr &)
```
Comment (chriadam) [inline src/incidencehandler.cpp]: unrelated, squashes a warning about unused parameters etc

File: src/incidencehandler.cpp [hunk 3]
```diff
@@ -193,17 +193,30 @@ void IncidenceHandler::copyIncidenceProperties(KCalCore::Incidence::Ptr dest, co
...
-        in >> *dest.data();
+        dest->recurrence()->clear();
+
+        KCalCore::Recurrence *dr = dest->recurrence();
+        KCalCore::Recurrence *sr = src->recurrence();
+
+        // recurrence rules and dates
+        KCalCore::RecurrenceRule::List srRRules = sr->rRules();
+        for (QList<KCalCore::RecurrenceRule*>::const_iterator it = srRRules.constBegin(), end = srRRules.constEnd(); it != end; ++it) {
+            KCalCore::RecurrenceRule *r = new KCalCore::RecurrenceRule(*(*it));
```
Comment (chriadam) [inline src/incidencehandler.cpp]: kcal documentation is atrocious, but addRRule takes ownership, so if we don't use operator new here (but instead just do:
    dr->addRRule(*it);
we eventually get a crash.

**Comment by pvuorela** [comment]
Has a bunch of manual steps in copying the recurrence, but assignment operator does look a bit suspicious. LGTM.

**Comment by chriadam** [comment]
Yeah, I couldn't get the assignment operator to work without crashing later on.  This code seems to work as is.

**Pull request merged**